### PR TITLE
Collection view data source snapshot functions

### PR DESCRIPTION
**Pull request by MartinJibber**
Creating convenience functions for updating the collection view datasource's data.

### CODE REVIEW

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 1]
```diff
@@ -14,6 +14,7 @@ import Foundation
 class CollectionViewDataSource<SectionType: Hashable, ItemType: Hashable> {
 
     typealias DiffableDataSourceType = UICollectionViewDiffableDataSource<SectionType, ItemType>
+    typealias SnapshotType = NSDiffableDataSourceSnapshot<SectionType, ItemType>
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:17]: Type alias to make function declarations a bit more succinct.

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 2]
```diff
@@ -62,17 +63,23 @@ class CollectionViewDataSource<SectionType: Hashable, ItemType: Hashable> {
...
 extension CollectionViewDataSource {
 
-    func apply(_ snapshot: NSDiffableDataSourceSnapshot<SectionType, ItemType>,
+    // MARK: - Standard DataSource Functions
+
+    func apply(_ snapshot: SnapshotType, animatingDifferences: Bool = true) {
+        self.diffableDataSource.apply(snapshot, animatingDifferences: animatingDifferences, completion: nil)
+    }
+
+    func apply(_ snapshot: SnapshotType,
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:72]: Made a synchronous version of apply. Sometimes you don't have to worry about doing follow up work. With this function you can just fire and forget.

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 3]
```diff
@@ -93,13 +100,226 @@ extension CollectionViewDataSource {
     }
 }
 
+// MARK: - Snapshot Convenience Functions
+
+extension CollectionViewDataSource {
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:105]: A bunch of convenience functions for mutating the snapshot.

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 4]
```diff
@@ -93,13 +100,226 @@ extension CollectionViewDataSource {
...
+        }
+    }
+
+    func reloadSections(_ identifiers: [SectionType]) {
+        self.applyChanges { snapshot in
+            snapshot.reloadSections(identifiers)
+        }
+    }
+
+    // Asynchronous Functions
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:211]: Asynchronous versions of the above functions. Use these if you want to do something after the animation finishes.

File: iOS/Collection View Management /CollectionViewDataSource.swift [hunk 5]
```diff
@@ -93,13 +100,226 @@ extension CollectionViewDataSource {
     }
 }
 
+// MARK: - Snapshot Convenience Functions
+
+extension CollectionViewDataSource {
+
+    // Synchronous Functions
```
Comment (MartinJibber) [inline iOS/Collection View Management /CollectionViewDataSource.swift:107]: Synchronous versions can be "fire-and-forget". Note that none of these have completion handlers. Any completion handlers should be handled using async functions below.

**Review by Benjamindodgson — APPROVED** [review]

**Pull request merged**